src/platform/utilities/sso/index.js

```diff
@@ -25,19 +26,25 @@ export async function ssoKeepAliveSession() {
     // ttl is null, we can't determine if the user has a session or not
     localStorage.removeItem('hasSessionSSO');
   }
-  return authn;
+  return { type, authn };
 }
 
 export async function checkAutoSession(application = null, to = null) {
-  const authn = await ssoKeepAliveSession();
+  const { type, authn } = await ssoKeepAliveSession();
   if (hasSession() && hasSessionSSO() === false) {
     // explicitly check to see if the SSOe session is false, as it could also
     // be null if we failed to get a response from the SSOe server, in which
     // case we don't want to logout the user because we don't know
     logout('v1', 'sso-automatic-logout');
-  } else if (!hasSession() && hasSessionSSO() && !getForceAuth() && authn) {
-    const params = { inbound: 'true', authn };
-    login('custom', 'v1', application, to, params, 'sso-automatic-login');
+  } else if (!hasSession() && hasSessionSSO() && !getForceAuth() && type) {
+    // only attempt an auto login if the user is
+    // a) does not have a VA.gov session
+    // b) has an SSOe session
+    // c) is not required for forceAuth (meaning their environment has SSOe
+    //    enabled and they have not previously tried to login)
+    // d) we have a non empty type value from the keepalive call to login with
+    const params = pickBy({ inbound: 'true', authn }, identity);
```

Comment:
I believe `identity` is the second argument by default, but also fine if you're just being explicit.

Reply:
ah it is, or at least in 4.17.  thanks https://lodash.com/docs/4.17.15#pickBy